src/utils/config_utils/inc/aduc/config_utils.h

```diff
@@ -53,28 +55,68 @@ typedef struct tagADUC_AgentInfo
 
 typedef struct tagADUC_ConfigInfo
 {
-    char* schemaVersion;
+    int refCount; /**< A reference count for this object. */
+
+    JSON_Value* rootJsonValue; /**< The root value of the configuration. */
+
+    const char* schemaVersion;
 
     JSON_Array* aduShellTrustedUsers; /**< All the trusted users for ADU shell. */
 
-    char* manufacturer; /**< Device info manufacturer. */
+    const char* manufacturer; /**< Device info manufacturer. */
 
-    char* model; /**< Device info model. */
+    const char* model; /**< Device info model. */
 
-    char* edgegatewayCertPath; /**< Edge gateway certificate path */
+    const char* edgegatewayCertPath; /**< Edge gateway certificate path */
 
     ADUC_AgentInfo* agents; /**< Array of agents that are configured. */
 
     unsigned int agentCount; /**< Total number of agents configured. */
 
-    char* compatPropertyNames; /**< Compat property names. */
+    const char* compatPropertyNames; /**< Compat property names. */
 
-    char* iotHubProtocol; /**< The IotHub transport protocol to use. */
+    const char* iotHubProtocol; /**< The IotHub transport protocol to use. */
 
     unsigned int
         downloadTimeoutInMinutes; /**< The timeout for downloading an update payload. A value of zero means to use the default. */
+
+    const char* aduShellFolder; /**< The folder where ADU shell is installed. */
+
+    char* aduShellFilePath; /**< The full path to ADU shell binary. */
+
+    char* configFolder; /**< The folder where ADU stores its configuration. */
+
+    const char* dataFolder; /**< The folder where ADU stores its data. */
+
+    const char* downloadsFolder; /**< The folder where ADU stores downloaded payloads. */
+
+    const char* extensionsFolder; /**< The folder where ADU stores its extensions. */
+
+    char* extensionsComponentEnumeratorFolder; /**< The folder where ADU stores its component enumerator extensions. */
```

Comment:
Does this need to be const char* like the others or is it assigned after? Ok to ignore since this is a merge.

Reply:
Good find. I think that would be good. I'll see if I can create a task about it for config utils as it's outside the scope for this PR.